pg_shard.c

```diff
@@ -537,9 +539,17 @@ ErrorIfQueryNotSupported(Query *queryTree)
 				specifiesPartitionValue = true;
 			}
 		}
+
+		whereClause = ((queryTree->jointree == NULL) ?
```

Comment:
We _seldom_ (never, maybe?) use the ternary operator in our style. I'd say this is more idiomatic CitusDB style:

```c
FromExpr *joinTree = query->jointree;
if (joinTree != NULL && contain_volatile_functions(joinTree->quals))
{
    hasNonConstQualExprs = true;
}
```